skipping unused sender channels in fabric tensix mode

yugaoTT opened the pull request:
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/26792)

### Problem description
currently we still keep all the sender channels (for 1d s0, s1, s2, for 2d E W N S) in fabric router, even though the tensix extension already moved all the sender channels there. We need to properly skip the fabric router sender channels and set those channels to 0 buffer slots.

### What's changed
for 1D, skip the s1/s2 channels in router, since those are moved to tensix extension, and only keep s0 there for mux connection.
for 2D, skip the non-worker sender channels (worker channel is the channel for current direction), and keep only the worker channel on router for mux to connect.
Fix a bug in fabric router, where it checks sender channel 0 to see if all the sender channels have been serviced. this is not true for 2D case, and should use the direction to index into the is_channel_serviced array.

fill in reserved L1 with dispatch/worker/tensix connection info,
dispatch connection info is used by dispatch (relay mux) to connect to the fabric router
worker connection info is used by workers to connect to either mux or fabric router
tensix connection info is used by tensix(mux) to connect to the fabric router

<img width="979" height="344" alt="Screenshot 2025-08-29 at 3 33 29 PM" src="https://github.com/user-attachments/assets/7be95883-ff03-4711-a9a8-286c793ae912" />

### Checklist
- [ ] [All post commit] https://github.com/tenstorrent/tt-metal/actions/runs/17406693823
- [x] galaxy quick https://github.com/tenstorrent/tt-metal/actions/runs/17405375209
- [x] t3k unit https://github.com/tenstorrent/tt-metal/actions/runs/17406418281

yugaoTT committed:
removing unused sender channels in fabric tensix mode

yugaoTT committed:
add helper

yugaoTT committed:
clean up

SeanNijjar commented:
are the post-dateline VCs still kept separate for the time being?

yugaoTT commented:
> are the post-dateline VCs still kept separate for the time being?

yes, the dateline VC is only being built for ring/torus, tensix extension doesn't change that logic. but I didn't remove any sender channels for different dateline locations for mux, so for example for 1d ring, each mux will always have 3 sender channels, even the dateline connection doesn't use s2 channel

SeanNijjar reviewed (approved):
I think the key thing (seemingly missing) is to have tests that run with and without fabric extended to worker cores so we can make sure both pathways still work as intended.

SeanNijjar commented:
> > are the post-dateline VCs still kept separate for the time being?
> 
> yes, the dateline VC is only being built for ring/torus, tensix extension doesn't change that logic. but I didn't remove any sender channels for different dateline locations for mux, so for example for 1d ring, each mux will always have 3 sender channels, even the dateline connection doesn't use s2 channel

but will the mux for s2 target receiver channel r1?

yugaoTT committed:
add vc1 sender channel back to fabric router

yugaoTT committed:
rebase fix

yugaoTT committed:
use non-stateful api when tensix extension enabled

yugaoTT committed:
clean up

yugaoTT committed:
add comments

yugaoTT commented on tt_metal/fabric/erisc_datamover_builder.cpp:
```diff
@@ -112,6 +112,42 @@ static void configure_risc_settings(
+    return is_2D_routing ? direction : 0;
+}
+
+static void update_sender_channel_servicing(
```
added

yugaoTT commented on tt_metal/fabric/erisc_datamover_builder.cpp:
```diff
@@ -396,6 +439,28 @@ void FabricEriscDatamoverConfig::configure_buffer_slots_helper(
+                default_num_sender_buffer_slots,
+                default_num_receiver_buffer_slots);
+            // set default buffer slots.
+            num_sender_buffer_slots[target_channel] = default_num_sender_buffer_slots;
```
yeah I created an issue here: https://github.com/tenstorrent/tt-metal/issues/27698, would be addressed by the same clean up pr

yugaoTT commented on tt_metal/fabric/erisc_datamover_builder.cpp:
```diff
@@ -765,9 +838,12 @@ FabricEriscDatamoverConfig::FabricEriscDatamoverConfig(
 
     auto skip_current_sender_channel = [&](uint32_t idx) -> bool {
+        bool is_2D_routing = FabricContext::is_2D_topology(topology);
+        uint32_t target_channel = is_2D_routing ? options.direction : 0;
```
added

yugaoTT commented:
> > > are the post-dateline VCs still kept separate for the time being?
> > 
> > 
> > yes, the dateline VC is only being built for ring/torus, tensix extension doesn't change that logic. but I didn't remove any sender channels for different dateline locations for mux, so for example for 1d ring, each mux will always have 3 sender channels, even the dateline connection doesn't use s2 channel
> 
> but will the mux for s2 target receiver channel r1?

I fixed the issue with vc1 being on mux, and moved it back to router.

yugaoTT committed:
update golden
